src/main/java/org/acra/util/IOUtils.java

```diff
@@ -128,4 +129,36 @@ public static String streamToString(@NonNull InputStream input, Predicate<String
             safeClose(reader);
         }
     }
+	
+	/**
+     * Reads an InputStream into a string in an async way
+     *
+     * @param input  the stream
+     * @param filter should return false for lines which should be excluded
+     * @param limit the maximum number of lines to read (the last x lines are kept)
+     * @return the read string
+     * @throws IOException
+     */
+    @NonNull
+    public static String streamToStringAsync(@NonNull InputStream input, Predicate<String> filter, int limit) throws IOException {
+        final BufferedReader reader = new BufferedReader(new InputStreamReader(input), ACRAConstants.DEFAULT_BUFFER_SIZE_IN_BYTES);
+        final NonblockingBufferedReader nonBlockingReader = new NonblockingBufferedReader(reader);
+        try {
+            String line;
+            final List<String> buffer = limit == NO_LIMIT ? new LinkedList<String>() : new BoundedLinkedList<String>(limit);
+            long end = System.currentTimeMillis() + READ_TIMEOUT;
+            while ((System.currentTimeMillis() < end)) {
```

Comment:
And if a developer don't want any delay on main / exception thread, acra's work should be done in a separate thread. For the moment I use a helper method in which I use a Runnable and start a new thread; again, not sure if a queue should be use.